src/test/java/tennisclub/dao/UserDaoImplTest.java

```diff
@@ -0,0 +1,155 @@
+package tennisclub.dao;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.annotation.DirtiesContext;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Role;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class UserDaoImplTest {
+    @PersistenceContext
+    EntityManager manager;
+
+    @Autowired
+    private UserDao userDao;
+
+    @Test
+    @Transactional
+    void notNullValidation() {
+        User blankUser = new User();
+        assertThrows(DataIntegrityViolationException.class, () -> userDao.create(blankUser));
+
+        User userWithUsername = new User();
+        userWithUsername.setUsername("username666");
+        assertDoesNotThrow(() -> userDao.create(userWithUsername));
+    }
+
+    @Test
+    @Transactional
+    void checkAllFieldsPersisted() {
+        User createdUser = createUser("honza42", "hon@za.cz", "Honza", Role.USER);
+        User foundUser = userDao.findAll().get(0);
```

Comment:
I think that a single unit test should test a single functionality. In this case, the `create` method of the dao is being tested but to check it, you use `findAll` on the dao. So now the passing of the test depends not only on the correctness of the tested functionality but also on some unrelated functionality of the object under test. The same thing is in some other test cases as well. 

The solution would be to use an instance of `EntityManager` to retrieve the created user. But that would require us to figure out how to use it in the tests.

Reply:
The only think I can think of now (to be able to user `EntityManager`) is only through `EntityManagerFactory`, which actually worked for me. However, if it's the case though, we might have to rewrite the tests for `Court.class` as well 😱

Reply:
@bigno78 I have used `EntityManager` successfully to detach an entity in the test for `update()`. Let me have a look whether it could be successfully used in other tests as well.

Reply:
@paveltobias @bigno78 I was able to successfully use `EntityManager` in tests 😊 it works with using PersistenceContext annotation 👀 but dunno if that's the correct way of doing it 😱😱😱

Reply:
@Fuyunori In either PV260 or PA165 PersistenceContext annotation was used so I believe it cannot be really bad :)

Reply:
@mazanec202 Done! (See [this commit](https://github.com/Fuyunori/PA165-project/pull/28/commits/f1319193ebdf98f551256c036f49ec314a563377) if you have time).